### PR TITLE
FIX-16597: Removed Community string characters validation from SNMP service form to accept all the characters

### PR DESCRIPTION
**Pull request by KaustubhDhokte**
I talked to Josh who had added this validation because there was some problem in middleware back then accepting some characters. I suppose now this validation is no longer needed as I could not find any constraint/convention for Community string in RFC or anywhere on the web.

### CODE REVIEW

**Comment by jpaetzel** [comment]
I find it hard to believe that any character at all can be used in the community string.

**Comment by KaustubhDhokte** [comment]
Hi Josh, I searched over the web but could not find any constraint or convention regarding best/standard practices for community string. Please let me know if any character(s) you think that should be disallowed. I will make the changes accordingly.

**Comment by jpaetzel** [comment]
Well, using @ or : in a community string is a very bad idea.  I think we should limit it to A-Za-z0-9 as well as -, _ and .

**Comment by KaustubhDhokte** [comment]
Added back validation to accept only [a-zA-Z0-9-_.,] as suggested by Josh.

**Comment by jpaetzel** [comment]
Ok, we are just about there.

1) In an RE a '.' is a metacharacter that means "match any character"  Are you sure it's properly escaped in this RE to match a literal '.'?

2) When I used a ',' it was not meant to be a character we accepted, I was using it as a delimiter for the characters we could accept.  Please remove the ',' (comma) as an allowed character. I apologize for being unclear.

**Comment by KaustubhDhokte** [comment]
1) Yes it matches to a literal '.' and no longer remains a special characters as we are using set [] of allowed characters. It is mentioned in RE documentation - "Special characters lose their special meaning inside sets.". I have tested this to confirm, it works as expected.

2) Apologies for the confusion. I'll remove the comma as an allowed character in the next commit.

File: gui/services/forms.py [hunk 1]
```diff
@@ -839,9 +839,9 @@ def clean_snmp_community(self):
                 raise forms.ValidationError(_('This field is required.'))
             else:
                 return community
-        if not re.match(r'^[-_,.a-zA-Z0-9\s]+$', community):
+        if not re.match(r'^[-_.a-zA-Z0-9\s]+$', community):
             raise forms.ValidationError(
-                _(u"The community must contain only alphanumeric characters ,"
+                _(u"The community must contain only alphanumeric characters "
```
Comment (jceel) [inline gui/services/forms.py]: Error message could be more descriptive with regards to what characters are allowed (looks like -, _ and . are allowed in addition to alphanumerics)

Comment (jceel) [inline gui/services/forms.py]: Sorry - the error message is fine. I misread the diff.

**Pull request merged**